mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxView.kt

```diff
@@ -29,10 +32,12 @@ interface MvRxView : LifecycleOwner {
      * a lifecycle event (e.g. rotation) you should assign a consistent id. Likely this means you should save the id
      * in onSaveInstance state. The viewId will not be accessed until a subscribe method is called.
      * Accessing mvrxViewId before calling super.onCreate() will cause a crash.
-     *
-     * For an example implementation see [BaseMvRxFragment].
      */
-    val mvrxViewId: String
+    val mvrxViewId: String get() = when (this) {
+        is ViewModelStoreOwner -> ViewModelProvider(this).get(MvRxViewIdViewModel::class.java).mvrxViewId
+        else -> error("If your MvRxView is not a ViewModelStoreOwner, you must implement mvrxViewId " +
+                "and return a string that is unique to this view and persistant across its entire lifecycle.")
```

Comment:
for this case might it be nice to keep the `MvRxViewId.kt ` class to make it easier to manually save/restore the id?

Reply:
@elihart That wouldn't be enough on its own though. It needs the state restoration.